crates/humanode-runtime/src/evm_address.rs

```diff
@@ -0,0 +1,202 @@
+use sp_runtime::{traits::LookupError, MultiAddress};
+
+use super::*;
+
+/// The EVM-aware address lookup for [`MultiAddress`].
+///
+/// The [`MultiAddress`] is a type that supports both "native" account ID (as [`MultiAddress::Id`])
+/// and, among other things, EVM-like 20-byte addresses (as [`MultiAddress::Address20`]).
+///
+/// We utilize this property to allow using both "native" and 20-byte addresses in our chain.
+/// This type, being part of the Substrate itself, and not part of our custom codebase,
+/// should be well supported by the node RPC API clients (like Polkadot.js API/Apps and subxt),
+/// so we can expect good compatibility in the existing ecosystem (well, at least this is one of
+/// the arguments for using this type in theory - we'll see how it plays out).
+pub struct MultiLookup<T>(core::marker::PhantomData<T>);
+
+impl<T> StaticLookup for MultiLookup<T>
+where
+    T: frame_system::Config<AccountId = AccountId32> + pallet_evm_accounts_mapping::Config,
+{
+    type Source = MultiAddress<T::AccountId, ()>;
+
+    type Target = T::AccountId;
+
+    fn lookup(s: Self::Source) -> Result<Self::Target, frame_support::error::LookupError> {
+        match s {
+            // Pass "native" address directly as-is.
+            MultiAddress::Id(id) => Ok(id),
+            // Map 20-byte address to a native address via ethereum address mapping pallet,
+            // if the mapping exists.
+            // If the mapping does not exist, we still want to proceed without an error and map the
+            // address to something. Well, the way we do this now is invoking
+            // the [`pallet_evm::HashedAddressMapping::<BlakeTwo256>`].
```

Comment:
Would like to get understanding of allowing this. What about the future possible mapping of this `Address20` with another `AccountId`?